proxy/nasshp/nassh.go

```diff
@@ -326,18 +326,26 @@ func (np *NasshProxy) allow(r *http.Request, w http.ResponseWriter, sid, hostpor
 		logid = LogId(sid, r, hostport, creds)
 	}
 	if np.filter != nil {
-		verdict := np.filter("tcp", hostport, creds)
-		if HostIp(hostport).IsUrl() {
-			for _, u := range HostIp(hostport).Resolve() {
+		host, port, err := net.SplitHostPort(hostport)
+		if err != nil {
+			goto DENY
+		}
+		{
+			res, err := net.LookupHost(host)
+			if err != nil {
+				goto DENY
```

Comment:
same comment as above. I would avoid the goto, in favor of a message that indicates what the problem is.
So eg, if DNS is broken, we notice.

(and it's possible for it to be broken: I had to ensure the proxy container used our hacked dns server for resolution, it was pointing to 8.8.8.8 originally)

Reply:
Done